### ISSUE TITLE
otlp/translator: Metrics with words split by multiple UTF8 character lose characters during translation

### ISSUE THREAD

**Issue by ArthurSens** [prometheus/prometheus#15362]
_Originally posted by @ArthurSens in https://github.com/prometheus/prometheus/pull/15258#discussion_r1833221478_

---
Due to historical reasons, the translator code was copied from OpenTelemetry collector codebase.
Over there, they tried to provide means to translate metric names following Prometheus conventions that are documented [here](https://prometheus.io/docs/practices/naming/).

Although not explicitly said, it was implied that words should be separated by a single underscore and the codebase was written with that in mind.

[Now that we're allowing OTel users to have their original names stored in prometheus without any transformation](https://github.com/prometheus/prometheus/pull/15258), we're facing problems where two (or more) UTF-8 characters are being used to separate words.

A few examples:
* `http______requests_total` becomes `http_requests_total`
* `http...._requests....total` becomes `http.requests.total`

In summary, we want to fix the bug documented in code [here](https://github.com/prometheus/prometheus/blob/03995775a7674a0f571abf667b82e07b9d48df24/storage/remote/otlptranslator/prometheus/normalize_name_test.go#L100-L116)
